python/lsst/pipe/base/configOverrides.py

```diff
@@ -139,8 +160,32 @@ class ConfigOverrides:
     necessary.
     """
 
-    def __init__(self):
-        self._overrides = []
+    def __init__(self) -> None:
+        self._overrides: list[tuple[OverrideTypes, Any]] = []
+        self._parameters: SimpleNamespace | None = None
+
+    def addParameters(self, parameters: ParametersIR) -> None:
+        """Add parameters which will be substituted when applying overrides.
+
+        Parameters
+        ----------
+        parameters : `ParametersIR`
+            Override parameters in the form as read from a Pipeline file.
+
+        Note
+        ----
+        This method may be called more than once, but each call will overwrite
+        any previous parameter defined with the same name.
+        """
+        match self._parameters:
+            case None:
+                namespace = self._parameters = SimpleNamespace()
```

Comment:
Why not just us `self._parameters` directly later, instead of adding the `namespace` local variable?